### PR TITLE
Property Manager - Review and Confirm, App & Reg Details pages

### PR DESCRIPTION
**Pull request by dimak1**
*Issue:*

- bcgov/entity/issues/23745
- bcgov/entity/issues/23746

*Description of changes:*
- Add Property Manager section to Review page
- Add Property manager section to Application Details and Registration Details pages
- Add unit tests
- Small fixes and minor tech debt cleanup

Screenshot 2024-10-25 at 09 57 11

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by dimak1** [comment]
Pending due to PR #209

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-210-f3c37s3c.web.app

**Comment by shaangill025** [comment]
> The fields in the detail pages are empty

@dimak1 , I believe adding `propertyManager` to `strr-web/interfaces/application-i.ts` will fix it [below].
```
export interface ApplicationDetailsI {
  registrationType: RegistrationTypeE,
  listingDetails: ListingDetailsI[]
  primaryContact: ContactI
  secondaryContact?: ContactI
  principalResidence: PrincipalResidenceI
  unitAddress: RegistrationAddressI
  propertyManager?: PropertyManagerI
  unitDetails: {
    parcelIdentifier?: string
    businessLicense?: string
    businessLicenseExpiryDate?: string
    propertyType: string
    ownershipType: string
  }
  documents?: DocumentUploadI[]
}
```

**Comment by dimak1** [comment]
Merged related PR. Rebased with main. Fixed issues and tests. Submission works, as well as Review, App Details and Reg Details pages. 
Pretty much got the Property Manager working. Please re-review. Thanks!
@shaangill025 @kris-daxiom @rstens

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-210-f3c37s3c.web.app

File: strr-web/interfaces/application-i.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export interface ApplicationDetailsI {
   primaryContact: ContactI
   secondaryContact?: ContactI
   principalResidence: PrincipalResidenceI
+  propertyManager: PropertyManagerI
```
Comment (shaangill025) [inline strr-web/interfaces/application-i.ts:47]: I think propertyManager should be optional

File: strr-web/interfaces/registration-i.ts [hunk 2]
```diff
@@ -9,31 +9,32 @@ export interface RegistrationAddressI {
...
-  principalResidence: PrincipalResidenceI,
-  sbc_account_id: number,
-  status: string,
-  submissionDate: string,
-  unitAddress: RegistrationAddressI,
+  listingDetails: { url: string }[]
+  primaryContact: ContactI
+  secondaryContact: ContactI | null
+  principalResidence: PrincipalResidenceI
+  propertyManager: PropertyManagerI
```
Comment (shaangill025) [inline strr-web/interfaces/registration-i.ts:25]: Same as above, propertyManager should be optional?

File: strr-web/interfaces/application-i.ts (hunk 1 above)
Comment (dimak1) [reply strr-web/interfaces/application-i.ts:47]: Good catch, yea it should be optional.

Comment (dimak1) [reply strr-web/interfaces/application-i.ts]: Updated

File: strr-web/interfaces/registration-i.ts (hunk 2 above)
Comment (dimak1) [reply strr-web/interfaces/registration-i.ts]: Updated

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://strr-ui-dev--pr-210-f3c37s3c.web.app

**Review by shaangill025 — APPROVED** [review]
LGTM

File: strr-web/components/bcros/form-section/contact-information/Form.vue [hunk 3]
```diff
@@ -203,13 +203,13 @@ onMounted(() => {
 const form = ref()
 
 watch(form, () => {
-  if (form.value && isComplete) { form.value.validate() }
+  if (form.value && isComplete) { form.value.validate({ silent: true }) }
```
Comment (dimak1) [inline strr-web/components/bcros/form-section/contact-information/Form.vue:206]: @shaangill025 this removed the validation error in console and fixed the issue with tests failures.

**Pull request merged**